src/Sql/Sql/Database/Cmdlet/NewAzureSqlDatabase.cs

```diff
@@ -198,14 +205,38 @@ public class NewAzureSqlDatabase : AzureSqlDatabaseCmdletBase<AzureSqlDatabaseCr
         /// </summary>
         [Parameter(Mandatory = false,
             HelpMessage = "The Backup storage redundancy used to store backups for the SQL Database. Options are: Local, Zone and Geo.")]
-        [ValidateSet("Local", "Zone", "Geo")]
+        [ValidateSet("Local", "Zone", "Geo", IgnoreCase = false)]
         public string BackupStorageRedundancy { get; set; }
 
+        protected static readonly string[] ListOfRegionsToShowWarningMessageForGeoBackupStorage = { "eastasia", "southeastasia", "brazilsouth", "east asia", "southeast asia", "brazil south" };
+
         /// <summary>
         /// Overriding to add warning message
         /// </summary>
         public override void ExecuteCmdlet()
         {
+            string location = ModelAdapter.GetServerLocation(ResourceGroupName, ServerName);
```

Comment:
- ModuleAdapter needs to be initialized before use( InitModelAdapter() is called in base.ExecuteCmdlet())
- ModelAdapter.GetServerLocation(ResourceGroupName, ServerName) will generate an additional call which requires re-recording of the related test cases.